### PR TITLE
Add stub @rules_python to third_party/

### PR DESCRIPTION
**Pull request by brandjon**
This stub contains only one relevant file, @rules_python//python:defs.bzl,
which mimics the file at the same path in bazelbuild/rules_python. Having this
repo gives us a way to inject this defs.bzl file into our protobuf dependency,
which is loaded as a separate (local) external repo and therefore cannot access
the existing //tools/python:private/defs.bzl in Bazel's own workspace. It also
means we'll be compatible with the future upstream migration to fix protobuf
for #9006.

A separate PR will add this to the Bazel root WORKSPACE file (since
third_party/ must be updated in a separate commit).

Break-out of #9019. Work toward #9006.

RELNOTES: None

### CODE REVIEW

File: third_party/rules_python/BUILD [hunk 1]
```diff
@@ -0,0 +1,5 @@
+filegroup(
+    name = "srcs",
+    srcs = glob(["**"]) + ["//third_party/rules_python/python:srcs"],
+    visibility = ["//visibility:public"],
```
Comment (katre) [inline third_party/rules_python/BUILD]: Limit the visibility to just //third_party:srcs. Same for python:srcs

Comment (brandjon) [reply third_party/rules_python/BUILD]: Done, let's see how CI likes it

File: third_party/rules_python/BUILD [hunk 2]
```diff
@@ -0,0 +1,5 @@
+filegroup(
+    name = "srcs",
+    srcs = glob(["**"]) + ["//third_party/rules_python/python:srcs"],
+    visibility = ["//third_party:srcs"],
```
Comment (katre) [inline third_party/rules_python/BUILD]: I meant:
visibility = ["//third_party:__pkg__"],
or similar.

**Comment by brandjon** [comment]
Update from offline discussions: Visibility is fixed in most recent commit but we're confirming CI passes. (Turns out my workstation failed bootstrapping tests because of low disk space.)

[Buildkite](https://buildkite.com/bazel/bazel-bazel/builds/9240) testing this commit without visibility fix (in addition to the one auto-triggered by this PR)

[Buildkite](https://buildkite.com/bazel/bazel-bazel/builds/9241) testing this commit with visibility fix

**Comment by brandjon** [comment]
[Passed](https://buildkite.com/bazel/bazel-bazel/builds/9241#7b9a5c0e-b3ff-4144-8042-00d2f67c2684) in manual CI run of branch containing this PR's content.

**Pull request closed without merging**